helm-chart/templates/deployment.yaml

```diff
@@ -68,18 +70,32 @@ spec:
               value: {{ .Values.environment.db_pwd}}
           ports:
             - name: http
-              containerPort: 80
+              containerPort: {{ .Values.container.port }}
               protocol: TCP
           livenessProbe:
             httpGet:
-              path: /webservices/status.php
+              path: /setup/
               port: http
           readinessProbe:
             httpGet:
-              path: /webservices/status.php
+              path: /setup/
               port: http
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+      initContainers:
+      - name: init-volume
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        command: [ 'sh', '-c', 'chown www-data:www-data conf data env-production log' ]
+        volumeMounts:
+        - mountPath: "/var/www/html/data"
+          name: itop-data-pv
+        - mountPath: "/var/www/html/env-production"
+          name: itop-env-pv
+        - mountPath: "/var/www/html/log"
+          name: itop-log-pv
+        - mountPath: "/var/www/html/conf"
+          name: itop-conf-pv
+
```

Comment:
Why is this needed? The directories already should have this owner and they are also already defined as volume by the Dockerfile.

Reply:
That must be some problem by your setup then:
```
$ docker run --rm supervisions/itop:3.0.2 ls -l
total 1128
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 addons
drwxr-xr-x  4 www-data www-data    4096 Sep 14 08:29 application
-rw-rw-r--  1 www-data www-data     952 Sep 12 14:08 approot.inc.php
-rw-rw-r--  1 www-data www-data    3308 Dec 21  2021 bootstrap.inc.php
drwxr-xr-x  3 www-data www-data    4096 Oct 20 14:55 conf
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 core
drwxr-xr-x  8 www-data www-data    4096 Sep 14 08:29 css
drwxr-xr-x  2 www-data www-data    4096 Oct 20 14:55 data
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 datamodels
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 dictionaries
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 env-production
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 extensions
drwxr-xr-x  9 www-data www-data    4096 Sep 14 08:29 images
-rw-rw-r--  1 www-data www-data    1368 Dec 21  2021 index.php
drwxr-xr-x 12 www-data www-data    4096 Sep 14 08:29 js
drwxr-xr-x 24 www-data www-data    4096 Sep 14 08:29 lib
drwxr-xr-x  2 www-data www-data    4096 Oct 20 14:55 log
-rw-rw-r--  1 www-data www-data 1046438 Sep 12 14:09 manifest.xml
drwxr-xr-x 14 www-data www-data    4096 Sep 14 08:29 node_modules
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 pages
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 portal
drwxr-xr-x  6 www-data www-data    4096 Sep 14 08:29 setup
drwxr-xr-x  8 www-data www-data    4096 Sep 14 08:29 sources
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 synchro
drwxr-xr-x  5 www-data www-data    4096 Sep 14 08:29 templates
-rw-rw-r--  1 www-data www-data     615 Dec 21  2021 web.config
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 webservices
```

Reply:
By default kubernetes mount volume as root

Reply:
www-data have the userid 33 not 1000.

Reply:
Okay, so a matching security context needs to be set instead of removing it I assume?

Reply:
I guess, In this case the container running as root and Apache is running as www-data.

Reply:
Yes, so if I understand well, I think all should be running fine when only `fsGroup` is set to `33`?

Reply:
I don't know the effect when you set only the fsgroup. 
The init container is really quick because pulling same image and juste running a chown. 
It's a good way to do that kind of action on containers :)

Reply:
Can you test so we are sure? I would prefer that instead of a workaround.

Reply:
Set `fsgroup`to 33 is not efficient please see : 
```
root@itop-itop-chart-b6c4f947f-jnpp8:/var/www/html# ls -l
total 1128
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 addons
drwxr-xr-x  4 www-data www-data    4096 Sep 14 08:29 application
-rw-rw-r--  1 www-data www-data     952 Sep 12 14:08 approot.inc.php
-rw-rw-r--  1 www-data www-data    3308 Dec 21  2021 bootstrap.inc.php
drwxrwsr-x  3 root     www-data    4096 Oct 24 08:08 conf
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 core
drwxr-xr-x  8 www-data www-data    4096 Sep 14 08:29 css
drwxrwsr-x  3 root     www-data    4096 Oct 24 08:08 data
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 datamodels
drwxr-xr-x  3 www-data www-data    4096 Sep 14 08:29 dictionaries
drwxrwsr-x  3 root     www-data    4096 Oct 24 08:08 env-production
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 extensions
drwxr-xr-x  9 www-data www-data    4096 Sep 14 08:29 images
-rw-rw-r--  1 www-data www-data    1368 Dec 21  2021 index.php
drwxr-xr-x 12 www-data www-data    4096 Sep 14 08:29 js
drwxr-xr-x 24 www-data www-data    4096 Sep 14 08:29 lib
drwxrwsr-x  3 root     www-data    4096 Oct 24 08:08 log
-rw-rw-r--  1 www-data www-data 1046438 Sep 12 14:09 manifest.xml
drwxr-xr-x 14 www-data www-data    4096 Sep 14 08:29 node_modules
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 pages
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 portal
drwxr-xr-x  6 www-data www-data    4096 Sep 14 08:29 setup
drwxr-xr-x  8 www-data www-data    4096 Sep 14 08:29 sources
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 synchro
drwxr-xr-x  5 www-data www-data    4096 Sep 14 08:29 templates
-rw-rw-r--  1 www-data www-data     615 Dec 21  2021 web.config
drwxr-xr-x  2 www-data www-data    4096 Sep 14 08:29 webservices

```

Reply:
But is that actually a problem? As it seems the group has the same permissions as the owner of this directory?

Reply:
Yes it is, itop is running as www-data and then the application can't write conf in these files

Reply:
I don't see the problem as user `www-data` is in the group `www-data` and that group can read and write to the log dir..

Reply:
Yes I'm just seeing this, let me do more test :)

Reply:
Updated with the `fsgroup`